Backend/ManagerLayer/UserManagement/UserManagementManager.cs

```diff
@@ -28,53 +35,111 @@ public int CreateUser(string email, string password, DateTime dob)
                 DateOfBirth = dob,
                 UpdatedAt = timestamp
             };
-            return _userService.CreateUser(user);
+
+            using (var _db = CreateDbContext())
+            {
+                var response = _userService.CreateUser(user, _db);
+                try
+                {
+                    return _db.SaveChanges();
+                }
+                catch (DbEntityValidationException ex)
+                {
+                    //catch error
+                    // detach user attempted to be created from the db context - rollback
+                    _db.Entry(response).State = System.Data.Entity.EntityState.Detached;
+                }
+            }
+            return 0;
         }
 
         public int DeleteUser(User user)
         {
             _userService = new UserService();
```

Comment:
User service should not be instantiated every time this method is called. It should only be instantiated in the constructor.